src/sys/unix/selector/kqueue.rs

```diff
@@ -304,7 +305,30 @@ impl Drop for Selector {
 }
 
 pub type Event = libc::kevent;
-pub type Events = Vec<Event>;
+pub struct Events(Vec<libc::kevent>);
+
+impl Events {
+    pub fn with_capacity(capacity: usize) -> Events {
+        Events(Vec::with_capacity(capacity))
+    }
+}
+
+impl Deref for Events {
+    type Target = Vec<libc::kevent>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for Events {
```

Comment:
And this `AsMut<[libc::kevent]>`?

Reply:
Nevermind, I'm wrong: as per an offline conversation, it seems like a lot more `Vec` methods _need_ to be called through `Events`, `Deref` is probably the right choice. Carry on.